community/lucene-index/src/main/java/org/neo4j/kernel/api/impl/schema/ValueEncoding.java

```diff
@@ -133,6 +133,72 @@ void setFieldValue( Value value, Field field )
                     field.setStringValue( value.prettyPrint() );
                 }
 
+                @Override
+                Query encodeQuery( Value value, int propertyNumber )
+                {
+                    return new ConstantScoreQuery(
+                            new TermQuery( new Term( key( propertyNumber ), value.prettyPrint() ) ) );
+                }
+            },
+    Spatial
+            {
+                @Override
+                String key()
+                {
+                    return "spatial";
+                }
+
+                @Override
+                boolean canEncode( Value value )
+                {
+                    return Values.isGeometryValue( value );
+                }
+
+                @Override
+                Field encodeField( String name, Value value )
+                {
+                    return stringField( name, value.prettyPrint() );
+                }
+
+                @Override
+                void setFieldValue( Value value, Field field )
+                {
+                    field.setStringValue( value.prettyPrint() );
+                }
+
+                @Override
+                Query encodeQuery( Value value, int propertyNumber )
+                {
+                    return new ConstantScoreQuery(
+                            new TermQuery( new Term( key( propertyNumber ), value.prettyPrint() ) ) );
+                }
+            },
+    Temporal
+            {
+                @Override
+                String key()
+                {
+                    return "temporal";
```

Comment:
Yes I had a chat with Johan about this and I'm quite sure all of the encodings are different:
`Date: 2018-04-06`
`LocalTime: 11:28:0.0`
`Time: 11:28:0.0+00:00`
`LocalDateTime: 2018-04-06T11:28:0.0`
`DateTime:2018-04-06T11:28:0.0+00:00`

I do't think it can be any clashes there.